### PR TITLE
copy: fix copying devices, fifos

### PR DESCRIPTION
**Pull request by tonistiigi**
fix for https://github.com/docker/buildx/issues/1027

Using FileMode as input to mknod was wrong. Also fix
handling of fifos. Socket copies make stub files.

@AkihiroSuda

### CODE REVIEW

File: copy/copy_nowindows.go [hunk 1]
```diff
@@ -26,3 +30,17 @@ func copyXAttrs(dst, src string, xeh XAttrErrorHandler) error {
...
+	st, ok := fi.Sys().(*syscall.Stat_t)
+	if !ok {
+		return errors.New("unsupported stat type")
+	}
+	var rDev int
+	if fi.Mode()&os.ModeDevice == os.ModeDevice || fi.Mode()&os.ModeCharDevice == os.ModeCharDevice {
+		rDev = int(st.Rdev)
+	}
+	mode := st.Mode
+	mode &^= syscall.S_IFSOCK // socket copied as stub
```
Comment (AkihiroSuda) [inline copy/copy_nowindows.go:44]: It might be good to have an option to toggle this behavior

**Review by AkihiroSuda — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
FreeBSD CI is failing
```
go: missing Git command. See https://golang.org/s/gogetcmd
error obtaining VCS status: exec: "git": executable file not found in $PATH
	Use -buildvcs=false to disable VCS stamping.
```

**Comment by tonistiigi** [comment]
https://github.com/tonistiigi/fsutil/pull/125

**Review by AkihiroSuda — APPROVED** [review]

**Comment by tonistiigi** [comment]
@AkihiroSuda Some updates were required after your review.

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**